DirectProgramming/DPC++FPGA/PlatformTests/board_test/src/board_test.hpp

```diff
@@ -0,0 +1,1852 @@
+    max_alloc_size =
+        q.get_device().get_info<info::device::max_mem_alloc_size>();
+    std::cout << "\nclGetDeviceInfo CL_DEVICE_GLOBAL_MEM_SIZE = "
+              << max_buffer_size << "\n";
+    std::cout << "clGetDeviceInfo CL_DEVICE_MAX_MEM_ALLOC_SIZE = "
+              << max_alloc_size << "\n";
+    std::cout << "Device buffer size available for allocation = "
+              << max_alloc_size << " bytes\n";
+  }
+
+  ~ShimMetrics() {}
+
+  size_t TestGlobalMem(queue &q);
+  int HostSpeed(queue &q);
+  int HostRWTest(queue &q, size_t dev_offset = 0);
+  int KernelClkFreq(queue &q, bool report_chk = true);
+  int KernelLatency(queue &q);
+  int KernelLaunchTest(queue &q);
+  int KernelMemRW(queue &q);
+  int KernelMemBW(queue &q);
+
+ private:
+  float h2d_rd_bw;
+  float h2d_wr_bw;
+  float h2d_rd_wr_bw;
+  bool h2d_rw_test;
+  float kernel_freq;
+  float kernel_latency;
+  float kernel_thruput;
+  float kernel_mem_bw;
+  bool kernel_mem_rw_test;
+  cl_ulong max_buffer_size;
+  cl_ulong max_alloc_size;
+};
+
+//////////////////////////////////////
+// **** TestGlobalMem function **** //
+//////////////////////////////////////
+
+// Input:
+// queue &q - queue to submit operation
+// Returns:
+// Number of errors in transfer OR 1 if device memory allocation is 0 or test
+// fails (return 0 means test passed)
+
+// The function does the following tasks:
+// 1. Get maximum device global memory allocation size
+// 2. Allocate host memory to use to store data to be written to and read from
+// device
+// 3. Write to device global memory
+// 4. Read from device global memory
+// 5. Verify data read back matches value written
+// 6. Report read, write bandwidth
+// If this test passes (returns 0), the host to device global memory interface
+// is working fine
+
+size_t ShimMetrics::TestGlobalMem(queue &q) {
+  // Data is transferred from host to device in kMaxHostChunk size transfers
+  // (size in bytes)
+  constexpr size_t kMaxHostChunk = 512 * kMB;
+
+  // Test fails if max alloc size is 0
+  if (max_alloc_size == 0) {
+    std::cout << "Maximum global memory allocation supported by Sycl device is "
+              << "0! Cannot run kernel-to-memory read wite test\n\n";
+    return 1;
+  }
+
+  // **** Create device buffer ****//
+  // Creating device buffer to span all usable global memory space on device
+  buffer<unsigned long, 1> dev_buf{
+      range<1>{(max_alloc_size / sizeof(unsigned long))}};
+  std::cout << "Size of buffer created = " << dev_buf.get_size() << " bytes\n";
+
+  // **** Host memory allocation **** //
+  /*<hostbuf> is used to store data
+  to be written to device buffer as well as data that is read back
+  While loop retries allocation for smaller chunk if kMaxHostChunk allocation
+  fails */
+  // Size of allocated host memory (in bytes)
+  size_t host_size = kMaxHostChunk;
+  unsigned long *hostbuf = (unsigned long *)malloc(host_size);
```

Comment:
Probably use `new` and `delete` rather than `malloc` and `free`

Reply:
done